### PR TITLE
Move mod/noscrape to Module\Noscrape

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #4090

### CODE REVIEW

File: src/Module/NoScrape.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	{
+		$a = DI::app();
+
+		$which = DI::args()->get(1);
+
+		$profile = 0;
+		if ((local_user()) && (DI::args()->get(2) === 'view')) {
+			$which   = $a->user['nickname'];
+			$profile = DI::args()->get(1);
+		}
```
Comment (MrPetovan) [inline src/Module/NoScrape.php]: Please replace this block by checking and using the `$parameters['profile']` variable.

File: src/Module/NoScrape.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ *
+ * @see GContact::updateFromNoScrape() for usage
+ */
+class NoScrape extends BaseModule
+{
+	public static function rawContent(array $parameters = [])
+	{
+		$a = DI::app();
+
+		$which = DI::args()->get(1);
```
Comment (MrPetovan) [inline src/Module/NoScrape.php]: Please replace this block by checking the existence and using the `$parameters['nick']` variable.

File: src/Module/NoScrape.php [hunk 3]
```diff
@@ -22,12 +22,17 @@ public static function rawContent(array $parameters = [])
...
+			// Get infos about a specific nick (public)
+			$which = $parameters['nick'];
+			$profile = 0;
+		} elseif (local_user() && isset($parameters['profile']) && DI::args()->get(2) == 'view') {
+			// view infos about a known profile (needs a login)
 			$which   = $a->user['nickname'];
-			$profile = DI::args()->get(1);
+			$profile = $parameters['profile'];
+		} else {
+			System::jsonError(404, 'Invalid request');
```
Comment (MrPetovan) [inline src/Module/NoScrape.php]: This will not happen as the routes won't be triggered for anything other than the two cases above.

Comment (nupplaphil) [reply src/Module/NoScrape.php]: It happens:
```console
localhost:~$ curl http://192.168.22.10/noscrape/friendica1/view
"Invalid request"
```
- valid route `/noscrape/{profile}/view`
- but not logged in

Comment (MrPetovan) [reply src/Module/NoScrape.php]: Oooh, this should be a error number 401 or a 403 in the `/noscrape/{profile}/view` case instead then, with a "Authentication required" or "Forbidden" or "You must be logged in" message.

**Pull request merged**